tests/test_batch.py

```diff
@@ -387,3 +391,69 @@ def test_get_batch_fetch_results_deprecated(
     ):
         with pytest.warns(DeprecationWarning):
             self.sdk.get_batch(self.batch_id, fetch_results=True)
+
+    @pytest.mark.parametrize(
+        "filters",
+        (
+            {"job_ids": [str(UUID(int=0x1))]},
+            {"start_date": datetime(2023, 1, 1)},
+            {"end_date": datetime(2023, 1, 1)},
+            {"status": "DONE"},
+            {"status": ["DONE", "PENDING"]},
+            {"min_runs": 10},
+            {"max_runs": 10},
+            OrderedDict(
+                job_ids=[str(UUID(int=0x1))],
+                status=["DONE", "PENDING"],
+                min_runs=10,
+                max_runs=10,
+                start_date=datetime(2023, 1, 1),
+                end_date=datetime(2023, 1, 1),
+            ),
+        ),
+    )
+    def test_rebatch_success(
+        self,
+        mock_request,
+        filters: Dict[str, Any],
+    ):
+        batch = self.sdk.rebatch(self.batch_id, **filters)
+        assert mock_request.last_request.method == "POST"
+
+        if ids := filters.pop("job_ids", None):
+            filters["id"] = ids
+            # Move id to beginning to construct query param string like the sdk does.
+            if isinstance(filters, OrderedDict):
+                filters.move_to_end("id", last=False)
+
+        # Constructing the query param string
+        query_params = "?"
+        for filter, value in filters.items():
+            if isinstance(value, list):
+                for val in value:
+                    query_params += f"{filter}={val}&"
+            else:
+                if isinstance(value, datetime):
+                    value = value.isoformat()
+                query_params += f"{filter}={value}&"
+        # Replace : by %3A for datetimes
+        query_params = query_params.replace(":", "%3A")[:-1]
```

Comment:
What happens here exactly ? I'm not sure I understood correctly

Reply:
: found in a date time is not a legal URL character instead it's replaced with %3a

Reply:
I investigated and yes it is the right conversion, good spot 👍🏻